test/jdk/java/awt/event/MouseEvent/ClickDuringKeypress/ClickDuringKeypress.java

```diff
@@ -22,115 +22,100 @@
+       } finally {
+           if (frame != null) {
+               EventQueue.invokeAndWait(frame::dispose);
+           }
+       }
+   }
+
+   static void createUI() {
       frame = new Frame("ClickDuringKeypress");
-      frame.addMouseListener(this);
+      frame.addMouseListener(clicker);
       frame.addWindowListener(new WindowAdapter() {
           public void windowActivated(WindowEvent e) {
-              synchronized(ClickDuringKeypress.this) {
+              synchronized(clicker) {
                   ready = true;
-                  ClickDuringKeypress.this.notifyAll();
+                  ClickDuringKeypress.clicker.notifyAll();
               }
           }
       });
       frame.setBounds(0, 0, 400, 400);
+    }
 
-      start();
-
-    }//End  init()
-
-   public void start ()
-    {
-      try {
-        robot = new Robot();
-      } catch (AWTException e) {
-        System.out.println("Could not create Robot.");
-        throw new RuntimeException("Couldn't create Robot.  Test fails");
-      }
-
-      robot.mouseMove(200, 200);
-      frame.show();
-
-      synchronized(this) {
-          try {
-              if (!ready) {
-                  wait(10000);
-              }
-          } catch (InterruptedException ex) {
-          }
-          if (!ready) {
-              System.out.println("Not Activated. Test fails");
-              throw new RuntimeException("Not Activated. Test fails");
-          }
+    static void doTest() throws Exception {
+       synchronized(clicker) {
+           try {
+               if (!ready) {
+                   clicker.wait(5000);
+               }
+           } catch (InterruptedException ex) {
+           }
+           if (!ready) {
+               System.out.println("Not Activated. Test fails");
+               throw new RuntimeException("Not Activated. Test fails");
+           }
       }
-
-      doTest();
-
-      //What would normally go into main() will probably go here.
-      //Use System.out.println for diagnostic messages that you want
-      //to read after the test is done.
-      //Use Sysout.println for messages you want the tester to read.
-
-    }// start()
-
-    // Mouse should be over the Frame by this point
-    private void doTest() {
+      // Mouse should be over the Frame by this point
       robot.setAutoDelay(2000);
       robot.waitForIdle();
       robot.keyPress(KeyEvent.VK_B);
-      robot.mousePress(InputEvent.BUTTON1_MASK);
+      robot.mousePress(InputEvent.BUTTON1_DOWN_MASK);
       robot.delay(10);
-      robot.mouseRelease(InputEvent.BUTTON1_MASK);
+      robot.mouseRelease(InputEvent.BUTTON1_DOWN_MASK);
       // Should trigger mouseClicked
       robot.keyRelease(KeyEvent.VK_B);
       robot.delay(1000);
 
       robot.setAutoDelay(DOUBLE_CLICK_AUTO_DELAY);
       for (int i = 0; i < CLICKCOUNT / 2; i++) {
-        robot.mousePress(InputEvent.BUTTON1_MASK);
+        robot.mousePress(InputEvent.BUTTON1_DOWN_MASK);
         robot.delay(10);
```

Comment:
Here and after
```suggestion
```

We have already called `robot.setAutoDelay(DOUBLE_CLICK_AUTO_DELAY);` before. We can set `DOUBLE_CLICK_AUTO_DELAY` to 20 and remove the multiple `robot.delay(10)` calls.

Probably we also want to add `setAutoWaitForIdle(true)`

Reply:
ok